src/arena.c

```diff
@@ -697,7 +767,13 @@ arena_reset(tsd_t *tsd, arena_t *arena) {
 		if (config_prof && opt_prof) {
 			prof_free(tsd, ptr, usize, &alloc_ctx);
 		}
-		large_dalloc(tsd_tsdn(tsd), edata);
+		if (config_prof && opt_prof && alloc_ctx.szind < nhbins &&
+		    unlikely(alloc_ctx.szind < SC_NBINS)) {
```

Comment:
nit: no need for the `unlikely` hint since this is a slow path

Reply:
Will fix. For context, this was just [copied directly from `arena_dalloc_large`](https://github.com/jemalloc/jemalloc/blob/b01d49664651f239fdf76774cb6de05ed7e63f4a/include/jemalloc/internal/arena_inlines_b.h#L301).